jenkins-dsl/jenkins_jira_integration.py

```diff
@@ -0,0 +1,352 @@
+#!/usr/bin/python
+
+import argparse
+import json
+import os
+import re
+import sys
+
+from typing import Dict, Set
+from jenkins import Jenkins
+from jira import JIRA
+
+import jenkins
+
+# Used in logging method to flip logging on and off
+VERBOSE = False
+
+# Some helpers to tidy up object indexing
+NO_PREVIOUS_BUILD = '-1'
+NUMBER = 'number'
+PREVIOUS_NUMBER = 'previous_number'
+UNKNOWN = 'unknown'
+
+
+def main() -> None:
+    """
+    TODO: Document once interface is stable
+    """
+    parser = argparse.ArgumentParser(description='Parse Jenkins build output and optionally update JIRA tickets with results for a single branch')
+    parser._action_groups.pop()
+    required = parser.add_argument_group('required arguments')
+    optional = parser.add_argument_group('optional arguments')
+
+    required.add_argument('--jenurl', type=str, help='Jenkins server url')
+    required.add_argument('--jenuser', type=str, required=True, help='JIRA server url')
+    required.add_argument('--jenpass', type=str, required=True, help='JIRA username')
```

Comment:
rather than pass in, can we read from a file?  Better security as command lines can be logged in plain text on the box running this command

Reply:
I've been running a shell script with credentials in it scoped to just my user account so things don't hit the history file at least. Is that materially different than parsing credentials from a file?

i.e. having something like `linkJenkinsToJira.sh <build_num>` that has the credentials stored in it scoped to just the owner of the file re: system permissions.

I thought about this a bit and, outside linux permissions on files, couldn't think of a way to cron something that required credentials without having some way to unwind them (key in .py script, plain text in file, etc). So it seemed like this was just as secure a solution as the alternatives, but that's depending on what you can pull from history with a shell script I suppose?